### DISCUSSION TITLE
Multidimensional sparse matrix

### DISCUSSION THREAD

**Discussion by hiramegl** [mathnet/mathnet-numerics#799] [Ideas]
Hi,

I've just started to use Math.Net and I wonder if is possible to create a multidimensional sparse matrix.
Even a dense matrix would work.

I tried Matrix &lt; Matrix &lt; double &gt; &gt; without success.

Is it possible to create matrices with more than 2 dimensions?

I would also like to operate the multidimensional matrix, like multiplication with a scalar or multiplication  with another matrix or vector.

Is it possible to do projections of a matrix?

Thanks for your help beforehand.

/Hiram

**Comment by cdrnet** [comment]
The matrices in Math.NET Numerics are really only matrices, not n-dimensional arrays. They are fixed to 2 dimensions and cannot be nested.

Unfortunately Math.NET Numerics does not provide any higher dimensional structures like tensors at this point.

**Comment by SRamezanH** [comment]
Hi,
I use somthing like this:
```
Matrix<float>[] Sigma;
Sigma = new Matrix<float>[10];
for (int i = 0; i < 10; i++) {
	Sigma[i] = Matrix<float>.Build.Dense(5, 5);
}
```
and it works well for me.
